pkg/backup/compaction_processor.go

```diff
@@ -349,9 +349,20 @@ func openSSTs(
 	endTime hlc.Timestamp,
 ) (mergedSST, error) {
 	var dirs []cloud.ExternalStorage
+	cleanupDirs := func(dirsToClose []cloud.ExternalStorage) {
+		for _, dir := range dirsToClose {
+			if err := dir.Close(); err != nil {
+				log.Warningf(ctx, "close export storage failed: %v", err)
+			}
+		}
+	}
+	// If we bail early due to an error, cleanup any external storage that was
```

Comment:
just so i understand the problem correctly: we have a memory leak if `openSSTs` bails early before passing back `mergedSST` to the caller, right?

Reply:
Yup that's correct